pkg/applier/stackapplier.go

```diff
@@ -55,57 +54,63 @@ func (s *StackApplier) Run(ctx context.Context) error {
 		return nil // The context is already done.
 	}
 
+	trigger := make(chan struct{}, 1)
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
 	watcher, err := fsnotify.NewWatcher()
 	if err != nil {
 		return fmt.Errorf("failed to create watcher: %w", err)
 	}
-	defer watcher.Close()
-
-	debounceCtx, cancelDebouncer := context.WithCancel(ctx)
-	defer cancelDebouncer()
-
-	debouncer := debounce.Debouncer[fsnotify.Event]{
-		Input:    watcher.Events,
-		Timeout:  1 * time.Second,
-		Filter:   s.triggersApply,
-		Callback: func(fsnotify.Event) { s.apply(debounceCtx) },
-	}
-
-	// Send an artificial event to ensure that an initial apply will happen.
-	go func() { watcher.Events <- fsnotify.Event{} }()
 
-	// Consume and log any errors.
 	go func() {
-		for {
-			err, ok := <-watcher.Errors
-			if !ok {
-				return
+		defer close(trigger)
+		defer func() {
+			if err := watcher.Close(); err != nil {
+				s.log.WithError(err).Error("Failed to close watcher")
 			}
-			s.log.WithError(err).Error("Error while watching stack")
-		}
+		}()
+		err = s.runWatcher(watcher, trigger, ctx.Done())
 	}()
```

Comment:
This part seems like unnecessary complication; `s.log` is available inside `s.runWatcher` so why not just do the defer and log the error inside `s.runWatcher` and just `go s.runWatcher`?